rust/worker/src/blockstore/types.rs

```diff
@@ -297,7 +300,17 @@ impl<
     pub(crate) async fn count(&'referred_data self) -> Result<usize, Box<dyn ChromaError>> {
         match self {
             BlockfileReader::MemoryBlockfileReader(reader) => reader.count(),
-            BlockfileReader::ArrowBlockfileReader(reader) => reader.count().await,
+            BlockfileReader::ArrowBlockfileReader(reader) => {
```

Comment:
would prefer this error handling happened at a lower level - we can just do reader.count().await if it underlying function returns something that conforms to Box<dyn ChromaError>. 

enum dispatch should be minimal